tests/benchdnn/common.cpp

```diff
@@ -243,12 +244,61 @@ static void zfree_protect(void *ptr) {
     ::free(ptr_start);
 }
 #endif
+struct memory_registry_t {
+    void add(void *ptr, size_t size) {
+        std::lock_guard<std::mutex> g(m_);
+        allocations_.emplace(std::pair<void *, size_t>(ptr, size));
+        total_size_ += size;
+
+        BENCHDNN_PRINT(6,
+                "[CHECKK_MEM]: zmalloc request with size %s, total "
+                "allocation size: %s\n",
+                smart_bytes(size).c_str(), smart_bytes(total_size_).c_str());
+        warn_size_check();
+    }
+    void remove(void *ptr) {
+        std::lock_guard<std::mutex> g(m_);
+        total_size_ -= allocations_[ptr];
+        allocations_.erase(ptr);
+    }
+
+    void set_expected_max(size_t size) {
+        expected_max_ = size;
+        warn_size_check();
+    }
+
+private:
+    size_t size() const { return total_size_; }
+    void warn_size_check() {
+        if (expected_max_ != unset_ && total_size_ > expected_max_) {
+            BENCHDNN_PRINT(1,
+                    "[CHECK_MEM][WARNING]: zmalloc allocation exceeded "
```

Comment:
If you plan to promote, I'd go with INFO instead, report all hits with current state and assign that task to Arch team. This will avoid unnecessary coloring CI while still be visible when the issue pops up.

I'd go with v0 since there's a high chance the run will be aborted because of amount misconduct.